src/topology/semicontinuous.lean

```diff
@@ -502,6 +503,16 @@ begin
   filter_upwards [h i y hi] with _ hx' using lt_supr_iff.2 ⟨i, hx'⟩,
 end
 
+lemma lower_semicontinuous_within_at_csupr {f : ι' → α → δ'}
+  (bdd : ∀ᶠ y in 𝓝[s] x, bdd_above (range $ λ i, f i y))
+  (h : ∀ i, lower_semicontinuous_within_at (f i) s x) :
+  lower_semicontinuous_within_at (λ x', ⨆ i, f i x') s x :=
+begin
+  assume y hy,
+  rcases exists_lt_of_lt_csupr hy with ⟨i, hi⟩,
+  filter_upwards [h i y hi, bdd] with y hy hy' using hy.trans_le (le_csupr hy' i)
+end
```

Comment:
You can deduce  `lower_semicontinuous_within_at_supr` using `lower_semicontinuous_within_at_csupr (by simp) h`. Same for other lemmas below.

Reply:
I don't think so because of the conditional version needs the index type to be nonempty

Reply:
Actually I just realized the result is trivially true when the indexing type is empty, so I can do that.

Reply:
Done!